cou/apps/app.py

```diff
@@ -74,8 +79,8 @@ def create(
         :rtype: Optional[OpenStackApplication]
         """
         # pylint: disable=too-many-arguments
-        if OpenStackCodenameLookup.is_charm_supported(charm):
-            if status.subordinate_to:
+        if is_charm_supported(charm):
+            if status.subordinate_to and charm not in SUBORDINATES:
```

Comment:
I don't think we need to check for `charm not in SUBORDINATES` here. If `is_charm_supported(charm)`, then we already know the charm is either principal or a known subordinate.

Reply:
hacluster charm is an example

Reply:
I was misled by the docstring change in `is_charm_version_supported` function. I though the lookup file only contains principal charms because of the word "primary". I suggestion changing `primary` to `core` in that function's docstring.

But still, I think this is a very confusing approach. `is_charm_supported == true` should indicate the charm is supported, so we should not need to have a condition afterwards to exclude some charms. I suggest either remove hacluster from the lookup file till we are ready to support it, or have a EXCLUDES list in openstack.py to exclude some charms and add `charm not in EXCLUDES` in `is_charm_supported` function.

Reply:
I fixed the docstring. We can talk the details.

Reply:
removed hacluster from csv file and no need for that block.